components/optolink.rst

```diff
@@ -87,6 +87,17 @@ Configuration variables:
 - **div_ratio** (*Optional*, int): Value factor of datapoint. Defaults to '1'.
 - All other options from :doc:`/components/sensor/index`
 
+.. warning::
+
+    Some temperature sensors might get negative values (e.g. Outside Temperature) but the data type is unsigned integer (positive only). You need to configure a sensor filter to convert it to a signed integer (positive and negative values possible). In such case don't configure ``div_ratio`` but use sensor filter instead. Example below:
+
+    .. code-block:: yaml
+
+        filters:
+          - lambda: !lambda |-
+              return (short)x;
+          - multiply: 0.1
+
```

Comment:
I know that support for negative values with `number` types is an issue (see https://github.com/j0ta29/esphome/issues/3). 
I didn't know that we have a similar issue with `sensor` types. Sensor values with more than 2 bytes are treated as float types and should therefore be handled correctly with negative value support. Can you please give me a full sensor configuration example where the default configuration without your workaround results in wrong values?!

Reply:
> Can you please give me a full sensor configuration example where the default configuration without your workaround results in wrong values?

Here is the sensor definition which does not work:
```
  - platform: optolink
    name: Outside Temperature
    address: 0x0800
    bytes: 2
    unit_of_measurement: °C
    device_class: temperature
    div_ratio: 1
    filters:
      - multiply: 0.1
    accuracy_decimals: 1
    update_interval: 66s
    state_class: measurement
    force_update: true
```
I use filter rather than the div_ratio as with div_ratio 10 I don't get decimals, which is nice to have for temperature sensors.

Reply:
Weirdly I do not have such a problem with my vitotronic 200 KW2, negative values are correctly reported for outside temperature, here is my conf :
```
- platform: optolink
  name: Outside temperature
  address: 0x0800
  bytes: 2
  div_ratio: 10
  accuracy_decimals: 1
  unit_of_measurement: °C
  device_class: temperature
  icon: "mdi:snowflake-thermometer"
  update_interval: 60s
```

Reply:
I'll give it a try when temperature drops again below zero (not in the forecast for the next 10 days). 
Maybe it is the `div_ratio` thing?

Reply:
Yes, a div_ratio of 1 leads the value to be treated as a integer without decimal places and ... unfortunately yet ... without support of negative values.
If you want to control decimal places use `accuracy_decimals` as @abuisine  recommended in his example.

Reply:
The temperatures dropped below zero now and inded this solution works. Many thanks!